### PR TITLE
Feature/red 122 student assessment overrides

### PR DESCRIPTION
**Pull request by red-tt-bear**

### CODE REVIEW

File: src/features/courses/course-route-validation.ts
Comment (PanopticaRising) [inline src/features/courses/course-route-validation.ts:106]: The problems you caught with us putting moments in the database didn't originate from Joi, right? If we hit these endpoints with moment from the frontend, everything is still fine?

File: src/features/courses/course-route-validation.ts
Comment (PanopticaRising) [inline src/features/courses/course-route-validation.ts:111]: Should these be explicitly optional?

File: src/features/courses/course-route-validation.ts
Comment (red-tt-bear) [reply src/features/courses/course-route-validation.ts:106]: * Correct, the issue @doombot-exe and I saw last night was the backend sending joi along
* I did not see an error on the frontend
* Furthermore a frontend type can't push through to the backend due to serialization and deserialization, if this was a problem joi date would have failed

**Review by PanopticaRising — APPROVED** [review]

File: src/features/courses/course-route-validation.ts
Comment (red-tt-bear) [reply src/features/courses/course-route-validation.ts:111]: They should be, however it doesn't matter for now, I wouldn't hold up the PR for it, once it's merged I'll fix in my current branch

**Pull request merged**
